tools/codegen/pkg/manifestmerge/filters.go

```diff
@@ -105,3 +131,133 @@ func featureGatesFromManifest(manifest metav1.Object) sets.String {
 	}
 	return ret
 }
+
+type ClusterProfileFilter struct {
+	clusterProfile string
+}
+
+func (f *ClusterProfileFilter) UseManifest(data []byte) (bool, error) {
+	partialObject := &metav1.PartialObjectMetadata{}
+	if err := kyaml.Unmarshal(data, partialObject); err != nil {
+		return false, err
+	}
+	// if there's no preferenceinclude everywhere
+	if !hasClusterProfilePreference(partialObject.GetAnnotations()) {
+		return true, nil
+	}
+
+	forThisProfile := partialObject.GetAnnotations()[f.clusterProfile] == "true"
+	return forThisProfile, nil
+}
+
+func (f *ClusterProfileFilter) UseCRD(metadata crdForFeatureSet) (bool, error) {
+	return metadata.clusterProfile == f.clusterProfile, nil
+}
+
+func (f *ClusterProfileFilter) String() string {
+	return fmt.Sprintf("clusterProfile=%v", f.clusterProfile)
+}
+
+type AndManifestFilter struct {
+	filters []ManifestFilter
+}
+
+func (f *AndManifestFilter) UseManifest(data []byte) (bool, error) {
+	for _, curr := range f.filters {
+		ret, err := curr.UseManifest(data)
+		if err != nil {
+			return false, err
+		}
+		if !ret {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func (f *AndManifestFilter) String() string {
+	str := []string{}
+	for _, curr := range f.filters {
+		str = append(str, fmt.Sprintf("%v", curr))
+	}
+	return strings.Join(str, " AND ")
+}
+
+type CRDFilter interface {
+	UseCRD(metadata crdForFeatureSet) (bool, error)
+}
+
+type AndCRDFilter struct {
+	filters []CRDFilter
+}
+
+func (f *AndCRDFilter) UseCRD(metadata crdForFeatureSet) (bool, error) {
+	for _, curr := range f.filters {
+		ret, err := curr.UseCRD(metadata)
+		if err != nil {
+			return false, err
+		}
+		if !ret {
+			return false, nil
+		}
+	}
+
+	return true, nil
+}
+
+func (f *AndCRDFilter) String() string {
+	str := []string{}
+	for _, curr := range f.filters {
+		str = append(str, fmt.Sprintf("%v", curr))
+	}
+	return strings.Join(str, " AND ")
+}
+
+type FeatureSetFilter struct {
+	featureSetName string
+}
+
+func (f *FeatureSetFilter) UseManifest(data []byte) (bool, error) {
+	partialObject := &metav1.PartialObjectMetadata{}
+	if err := kyaml.Unmarshal(data, partialObject); err != nil {
+		return false, err
+	}
+
+	forThisFeatureSet := partialObject.GetAnnotations()["release.openshift.io/feature-set"] == f.featureSetName
+	return forThisFeatureSet, nil
+}
+
+func (f *FeatureSetFilter) UseCRD(metadata crdForFeatureSet) (bool, error) {
+	return metadata.featureSet == f.featureSetName, nil
+}
+
+func (f *FeatureSetFilter) String() string {
+	return fmt.Sprintf("featureSetName=%v", f.featureSetName)
+}
+
+type HasData struct {
+}
+
+func (f *HasData) UseCRD(metadata crdForFeatureSet) (bool, error) {
+	return metadata.noData == false, nil
+}
+
+func (f *HasData) String() string {
+	return "HasData"
+}
+
+type Everything struct {
+}
+
+func (f *Everything) UseManifest(data []byte) (bool, error) {
+	return true, nil
+}
+
+func (f *Everything) UseCRD(metadata crdForFeatureSet) (bool, error) {
```

Comment:
Currently the error return appears unused on all cases of this filter, that I can see in this diff, is this a new interface?